third-party/realdds/include/realdds/dds-stream.h

```diff
@@ -16,41 +17,22 @@ namespace realdds {
 class dds_stream
 {
 public:
-    struct profile
-    {
-        int16_t uid;       // Stream unique ID
-        int16_t framerate; // FPS
-        int8_t type;       // Kind of video, motion etc..
-        int8_t format;     // Data format specification
-        int8_t index;      // Used to distinguish similar streams like IR L / R
-    };
-
-    virtual int8_t get_type() const = 0;
     virtual const std::string & get_group_name() const = 0;
 
-    virtual void add_profile( const dds_stream::profile & prof, bool default_profile ) = 0;
-    virtual size_t foreach_profile( std::function< void( const dds_stream::profile & prof, bool def_prof ) > fn ) const = 0;
+    virtual size_t foreach_profile( std::function< void( const dds_stream_profile & prof, bool def_prof ) > fn ) const = 0;
```

Comment:
This should pass the shared_ptr.
And while at it, we should remove the `def_prof` argument and replace it with a `default_profile()` function that returns the default.

Reply:
Agreed. For next PR